### PR TITLE
Picking Consensus and Remove Duplicates

### PR DESCRIPTION
**Pull request by DavidHerreros**
Port of protocols `protocol_particle_pick_remove_duplicates` and `protocol_particle_pick_consensus` from Xmipp to work with Tomo objects `SetOfCoordinates3D` and `Coordinate3D`.

Together with the protocols a couple of tests have been also included (`TestTomoPickingConsenus` and `TestTomoPickingRemoveDuplicates`).

### CODE REVIEW

File: tomo/protocols/protocol_particle_pick_consensus.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+        form.addSection(label='Input')
+        form.addParam('inputCoordinates', params.MultiPointerParam,
+                      pointerClass='SetOfCoordinates3D',
+                      label="Input 3D coordinates", important=True,
+                      help='Select the set of 3D coordinates to compare')
+        form.addParam('consensusRadius', params.IntParam, default=10,
+                      label="Radius",
+                      help="All coordinates within this radius (in pixels) "
+                           "are presumed to correspond to the same particle")
+        form.addParam('consensus', params.IntParam, default=-1,
```
Comment (EstrellaFG) [inline tomo/protocols/protocol_particle_pick_consensus.py:86]: I think is more clear for the user if this param is a `EnumParam` with `display=EnumParam.DISPLAY_HLIST` and options are directly 'AND'/'OR' in the form. Just a sugerence :)

File: tomo/protocols/protocol_particle_pick_consensus.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+                      label="Consensus",
+                      help="How many times need a particle to be selected to "
+                           "be considered as a consensus particle.\n"
+                           "*Set to -1* to indicate that it needs to be selected "
+                           "by all algorithms: *AND* operation.\n"
+                           "*Set to 1* to indicate that it suffices that only "
+                           "1 algorithm selects the particle: *OR* operation.")
+        form.addParam('mode', params.EnumParam, label='Consensus mode',
```
Comment (EstrellaFG) [inline tomo/protocols/protocol_particle_pick_consensus.py:94]: I think `display=EnumParam.DISPLAY_HLIST` here works better too. Just another small sugerence.

**Review by EstrellaFG — APPROVED** [review]
Very usefull protocols! Just small sugerences.

File: tomo/protocols/protocol_particle_pick_consensus.py (hunk 2 above)
Comment (DavidHerreros) [reply tomo/protocols/protocol_particle_pick_consensus.py:94]: True! HLIST display mode is much intuitive. Changed done!

File: tomo/protocols/protocol_particle_pick_consensus.py (hunk 1 above)
Comment (DavidHerreros) [reply tomo/protocols/protocol_particle_pick_consensus.py:86]: In principle, the AND/OR are just two specific cases for the consensus parameter. In fact, if you have N tomograms you can place any number between -1 and N to determine how many times the coordinate has to appear in the Tomograms to be considered in the consensus (the cases for -1, 0 and N will do the same).

Comment (EstrellaFG) [reply tomo/protocols/protocol_particle_pick_consensus.py:86]: Ok then, I thought that it were just two options.

Comment (DavidHerreros) [reply tomo/protocols/protocol_particle_pick_consensus.py:86]: It is true that it is not so clear... I will improve the help so it covers better all the options of the parameter :)

**Review by JorMaister — APPROVED** [review]

**Pull request merged**